### ISSUE TITLE
Overwatch: Battle.net Launcher doesn't start starting wine-staging at commit 02913f7

### ISSUE THREAD

**Issue by AdelKS** [lutris/lutris#2991]
Hello! First of all, thank you for your amazing work!

I am here to report what seems to be a bug and I am ready to try to fix it myself if I get enough help. Since I don't have enough knowledge about wine's shenanigans.

## Bug description

Starting from commit 02913f7 of the [wine-staging](https://github.com/wine-staging/wine-staging) code, the Battle.net Launcher from Overwatch doesn't seem to start anymore from Lutris (which enables then to run Overwatch).

I have confirmed at least that if I go to the wine executable from that specific wine-staging version that doesn't work with Lutris and use it to run Battle.net with 
```shell

/wine-staging-02913f7/wine "/path/to/overwatch/drive_c/Program Files (x86)/Battle.net/Battle.net.exe"

```

Battle.net does work: I can login and launch the game (without dxvk)

## Steps to reproduce

- Clone TK-Glitch's git to compile wine-staging with fsync : https://github.com/Frogging-Family/wine-tkg-git
- Open the file ``wine-tkg-git/customization.cfg``
  - Update ``_nomakepkg_dep_resolution_distro`` with the used distro
   - reference wine-staging's commit : ``_staging_version="02913f7"``
- Compile by running `wine-tkg-git/non-makepkg-build.sh`
- Install Overwatch in Lutris on a "regular" wine version (the game won't be downloaded, it will only take few minutes)
- Verify that the Battle.net Launcher works fine with the regular wine version
- Add the custom compiled wine to Lutris and try use it to run Overwatch

Even though I am using wine-staging with additional patches, I am sure the problem doesn't come from the patchset that is applied on top of wine-staging. Because that same patchset is applied on earlier wine-staging versions and the launcher works fine. But if you wish I can try vanilla wine-staging at that commit and report back.

## Debug information

[lutris-debug-out.txt](https://github.com/lutris/lutris/files/4874204/lutris-debug-out.txt)

[lutris-issue-report.txt](https://github.com/lutris/lutris/files/4874206/lutris-issue-report.txt)

**Comment by byte1024** [comment]
Is this yet another messed up mangohud situation lol?

> **Game is considered started.**
> fsync: up and running.
> **MANGOHUD: Failed to open 64bit libnvidia-ml.so.1: libnvidia-ml.so.1: cannot open shared object file: No such file or directory**
> WARNING: radv is not a conformant vulkan implementation, testing use only.
> Initial process has exited.
> **Game is considered exited.**

**Comment by AdelKS** [comment]
Those seem to be soft fails and always happen for me. MangoHud still works even with those warnings.

**Comment by byte1024** [comment]
It looks like that commit was tagged for 5.11/5.12 so maybe applied too early without other commits? Wine staging 5.10 as supplied normally works fine - I just tested it.

> wine-tkg-staging-fsync-git-5.10.r13.g02913f75/bin/win

Another note is that a Lutris commit was just added to disable fsync and I don't know if that also could be part of the problem if you run Lutris with it enabled. For some reason I'm not seeing a Lutris version here.

I have too many questions now  lol.

**Comment by AdelKS** [comment]
> It looks like that commit was tagged for 5.11/5.12 so maybe applied too early without other commits?

Wine-staging commits aren't cherry picked. The script clones wine-staging and checks out a certain commit, then applies other patches on top of it. 

> Wine staging 5.10 as supplied normally works fine - I just tested it.

Yes that works for me too, it's starting commit 02913f7, who's between 5.10 and 5.11, that Battle.net doesn't start anymore.

**Comment by byte1024** [comment]
In that case I will just assume that the lutris wine versions are still problematic. That is what I read as to why they're not showing up now with 5.7 being the latest in the wine runner list.

I just don't have the desire to poke directly into their wine git any further.

If they're having problems then it will be sorted in time.

**Comment by AdelKS** [comment]
> In that case I will just assume that the lutris wine versions are still problematic. That is what I read as to why they're not showing up now with 5.7 being the latest in the wine runner list.
...
If they're having problems then it will be sorted in time.

Sounds reasonable. Though I was a bit expecting that Lutris isn't "Runner dependent". Maybe someone can clarify this for me.

> If it works for you without it then I say go for it :-)

I may take this road, but it seems to be a pain to set by hand all the correct environment variables before launching the game directly with wine so it uses dxvk and mangohud.

> And if its a regression in wine staging formal then test and find out I suppose. And if it runs there then its tkg.

It doesn't seem to be, since I can start the Battle.net launcher directly with wine, with that same versions that doesn't work with lutris. It is possible that I should just do some kind of environment cleanup for it to work. I couldn't find where to do that.

**Comment by byte1024** [comment]
Oh, I just noticed that you're literally using tkg-glitch's wine there. You know he doesn't support Lutris anymore right? Probably be best to post an issue with his git, though I could see him closing it if you mention Lutris.

The Lutris one is here https://github.com/lutris/wine and seems to have nothing newer than 5.7 for commits.

**Comment by AdelKS** [comment]
I haven't tried wine-staging vanilla at that specific commit, but I am pretty sure the same issue will happen. I will do later to check.

In another hand, isn't Lutris a wrapper that properly defines env variables for dxvk/wine/other things before launching a game ?

**Comment by byte1024** [comment]
Yes, it looks like they're still trying to use tkg's wine so it should work better without applying the extra stuff on top as it does for you. Which makes me think that they could do a basic version which would at least supply tkg's and keep the versions increasing.

**Comment by strycore** [comment]
I'm moving this discussion to `lutris/wine` since if a fix is needed, it will be there.
Also please test with our latest 3.6 or 3.7 builds and see if the problem happens.
